src/object/obj_ec.h

```diff
@@ -741,8 +772,16 @@ obj_ec_parity_lists_match(struct daos_recx_ep_list *lists_1,
 	for (i = 0; i < nr; i++) {
 		list_1 = &lists_1[i];
 		list_2 = &lists_2[i];
-		if (list_1->re_nr != list_2->re_nr ||
-		    list_1->re_ep_valid != list_2->re_ep_valid) {
+		if (list_1->re_ep_valid != list_2->re_ep_valid) {
+			D_ERROR("got different ep_valid in EC data recovery\n");
+			return -DER_IO;
+		}
+		if (list_1->re_nr != list_2->re_nr) {
```

Comment:
need compare the list_1 and list_2 even list_1->re_nr == list_2->re_nr ?

Reply:
I think need not compare that case, when the re_nr equals, the two recx lists should be match exactly.

Reply:
ah, it will compare in the following.